src/main/java/fr/cnes/sonarqube/plugins/framac/sensor/FramaCSensor.java

```diff
@@ -0,0 +1,277 @@
+        final ActiveRules activeRules = sensorContext.activeRules();
+
+        // run Frama-C execution
+        if(config.getBoolean(FramaCPluginProperties.AUTOLAUNCH_PROP_KEY).orElse(Boolean.getBoolean(FramaCPluginProperties.AUTOLAUNCH_PROP_DEFAULT))) {
+            executeFramaC(sensorContext, config);
+        }
+
+        // Report files found in file system and corresponding to SQ property.
+        final List<String> reportFiles = getReportFiles(config, fileSystem);
+
+        // If exists, unmarshal each xml result file.
+        for(final String reportPath : reportFiles) {
+
+            // Analyse CSV report
+            final List<FramaCError> errors = analyseReportCsv(reportPath);
+            // Retrieve file in a SonarQube format.
+            final Map<String, InputFile> scannedFiles = getScannedFiles(fileSystem, errors);
+
+            // Handles issues.
+            for (final FramaCError issue : errors) {
+                if (isRuleActive(activeRules, issue.getRuleKey())) { // manage active rules
+                    saveIssue(sensorContext, scannedFiles, issue);
+                } else { // log ignored data
+                    LOGGER.info(String.format(
+                            "An issue for rule '%s' was detected by Frama-C but this rule is deactivated in current analysis.",
+                            issue.getRuleKey()));
+                }
+            }
+        }
+
+    }
+
+    /**
+     * Execute Frama-C through a system process.
+     *
+     * @param sensorContext Context of the sensor.
+     * @param config Configuration of SonarQube instance.
+     */
+    private void executeFramaC(final SensorContext sensorContext, final Configuration config) {
+        LOGGER.info("Frama-C auto-launch enabled.");
+        final String executable = config.get(FramaCPluginProperties.FRAMAC_PATH_KEY).orElse(FramaCPluginProperties.FRAMAC_PATH_DEFAULT);
+        final String[] files = sensorContext.fileSystem().baseDir().list();
+        final String outputFile = config.get(FramaCPluginProperties.REPORT_PATH_KEY).orElse(FramaCPluginProperties.REPORT_PATH_DEFAULT);
+        final String outputPath = Paths.get(sensorContext.fileSystem().baseDir().toString(),outputFile).toString();
+        final String outputOption = "-o";
+        final String command = String.join(" ", executable, String.join(" ",files), outputOption, outputPath);
+
+        LOGGER.info("Running Frama-C and generating results to "+ outputPath);
+        try {
+            final Process framac =  Runtime.getRuntime().exec(command);
+            int success = framac.waitFor();
+            if(0!=success){
+                final String message = String.format("Frama-C auto-launch analysis failed with exit code %d.",success);
+                throw new FramaCException(message);
+            }
+            LOGGER.info("Auto-launch successfully executed Frama-C.");
+        } catch (InterruptedException | IOException | FramaCException e) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Either re-interrupt this method or rethrow the "InterruptedException". ([squid:S2142](https://sonarcloud.io/organizations/lequal/rules#rule_key=squid%3AS2142))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fr.cnes.sonarqube.plugins%3Asonar-frama-c-plugin&pullRequest=20&issues=AWXs1bJoxJR_uy9B9CnU&open=AWXs1bJoxJR_uy9B9CnU)

[ISSUE_KEY]: AWXs1bJoxJR_uy9B9CnU